src/components/inputs/rich-text-input/editor.js

```diff
@@ -0,0 +1,347 @@
+import {
+  Toolbar,
+  ToolbarMainControls,
+  ToolbarRightControls,
+  EditorContainer,
+  Container,
+} from './rich-text-input.styles';
+import Divider from './divider';
+import MultiDropdown from './multi-dropdown';
+import messages from './messages';
+
+const tooltipStyles = {
+  body: {
+    zIndex: 9999,
+  },
+};
+const DEFAULT_NODE = 'paragraph';
+
+const NUMBERED_LIST = 'numbered-list';
+const BULLETED_LIST = 'bulleted-list';
+const LIST_ITEM = 'list-item';
+
+const COLLAPSED_HEIGHT = 64;
+
+const dropdownOptions = [
+  { label: 'Strikethrough', value: 'strikethrough' },
+  { label: 'Superscript', value: 'superscript' },
+  { label: 'Subscript', value: 'subscript' },
+];
+
+const Editor = props => {
+  const intl = useIntl();
+  const { editor } = props;
+  const ref = React.useRef();
+  const [renderToggleButton, setRenderToggleButton] = React.useState(false);
+
+  React.useEffect(() => {
+    if (ref.current.clientHeight > COLLAPSED_HEIGHT && !renderToggleButton) {
+      setRenderToggleButton(true);
+    }
+    if (ref.current.clientHeight <= COLLAPSED_HEIGHT && renderToggleButton) {
+      setRenderToggleButton(false);
+    }
+  }, [editor.value, renderToggleButton]);
+
+  const activeBlock =
+    (editor.value.blocks.first() && editor.value.blocks.first().type) || '';
+
+  const activeMarks = Array.from(editor.value.activeMarks).map(
+    mark => mark.type
+  );
+
+  const activeMoreStyleMarks = activeMarks.filter(activeMark =>
+    dropdownOptions.some(dropdownOption => activeMark === dropdownOption.value)
+  );
+
+  const hasBlock = type => editor.value.blocks.some(node => node.type === type);
+
+  const onClickBlock = ({ value: type }) => {
+    const { value } = editor;
+    const { document } = value;
+
+    // Handle everything but list buttons.
+    if (type !== BULLETED_LIST && type !== NUMBERED_LIST) {
+      const isActive = hasBlock(type);
+      const isList = hasBlock(LIST_ITEM);
+
+      if (isList) {
+        editor
+          .setBlocks(isActive ? DEFAULT_NODE : type)
+          .unwrapBlock(BULLETED_LIST)
+          .unwrapBlock(NUMBERED_LIST);
+      } else {
+        editor.setBlocks(isActive ? DEFAULT_NODE : type);
+      }
+    } else {
+      // Handle the extra wrapping required for list buttons.
+      const isList = hasBlock(LIST_ITEM);
+      const isType = value.blocks.some(block => {
+        return !!document.getClosest(block.key, parent => parent.type === type);
+      });
+
+      if (isList && isType) {
+        editor
+          .setBlocks(DEFAULT_NODE)
+          .unwrapBlock(BULLETED_LIST)
+          .unwrapBlock(NUMBERED_LIST);
+      } else if (isList) {
+        editor
+          .unwrapBlock(type === BULLETED_LIST ? NUMBERED_LIST : BULLETED_LIST)
+          .wrapBlock(type);
+      } else {
+        editor.setBlocks(LIST_ITEM).wrapBlock(type);
+      }
+    }
+  };
+
+  const { toggleMark } = editor;
+
+  const onChangeMoreStyles = React.useCallback(
```

Comment:
Is there a reason we `useCallback` here but not above?

Reply:
going to remove these to make it simpler